worlds/animal_well/client.py

```diff
@@ -170,18 +184,105 @@ def on_package(self, cmd: str, args: dict):
         if cmd == "Connected":
             self.slot_data = args.get("slot_data", {})
 
+        try:
+            def parse_ids(data):
+                for item in data:
+                    if 'type' in item and 'text' in item:
+                        if item['type'] == 'player_id':
+                            item['text'] = self.player_names[int(item['text'])] if item['text'].isnumeric() else item['text']
+                        elif item['type'] == 'item_id':
+                            item['text'] = item_id_to_name[int(item['text'])] if item['text'].isnumeric() else item['text']
```

Comment:
any reason not to use ctx.item_names.lookup_in_slot/game?
benefits are 1) when/if you change the datapackage you don't have to force apworld updates for all clients 2) you can handle non-local IDs (like location text in hints)
also FYI [raw_text_parser](https://github.com/ArchipelagoMW/Archipelago/blob/main/CommonClient.py#L361) exists, not certain the usecase overlaps fully

Reply:
I'll give those a shot.

Reply:
Replaced parseIds with using the appropriate functions to find item and location names